### PR TITLE
SDI-613 Add integration tests

### PR DESCRIPTION
**Pull request by mikehalmamoj**
Disabled for now but help to define what the amend activity service should do

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisTest.kt [hunk 1]
```diff
@@ -15,161 +25,216 @@ import uk.gov.justice.digital.hmpps.prisonertonomisupdate.wiremock.MappingExtens
 import uk.gov.justice.digital.hmpps.prisonertonomisupdate.wiremock.NomisApiExtension.Companion.nomisApi
 import uk.gov.justice.hmpps.sqs.countAllMessagesOnQueue
 
-private const val activityScheduleId: Long = 100
-private const val activityId: Long = 200
-private const val courseActivityId: Long = 300
-private const val allocationId: Long = 400
-private const val bookingId: Long = 500
+private const val ACTIVITY_SCHEDULE_ID: Long = 100
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisTest.kt:28]: Fix Intellij warning

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisTest.kt [hunk 2]
```diff
@@ -15,161 +25,216 @@ import uk.gov.justice.digital.hmpps.prisonertonomisupdate.wiremock.MappingExtens
...
 
-  @Test
-  fun `will consume a create activity schedule message`() {
-    activitiesApi.stubGetSchedule(activityScheduleId, buildApiActivityScheduleDtoJsonResponse())
-    activitiesApi.stubGetActivity(activityId, buildApiActivityDtoJsonResponse())
-    mappingServer.stubGetMappingGivenActivityScheduleIdWithError(activityScheduleId, 404)
-    mappingServer.stubCreateActivity()
-    nomisApi.stubActivityCreate("""{ "courseActivityId": $courseActivityId }""")
+  @Nested
+  inner class CreateActivitySchedule {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisTest.kt:37]: Group related tests into classes.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisTest.kt [hunk 3]
```diff
@@ -15,161 +25,216 @@ import uk.gov.justice.digital.hmpps.prisonertonomisupdate.wiremock.MappingExtens
...
+      activitiesApi.stubGetActivity(ACTIVITY_ID, buildApiActivityDtoJsonResponse())
+      mappingServer.stubGetMappingGivenActivityScheduleIdWithError(ACTIVITY_SCHEDULE_ID, 404)
+      mappingServer.stubCreateActivity()
+      nomisApi.stubActivityCreate("""{ "courseActivityId": $COURSE_ACTIVITY_ID }""")
 
-    await untilCallTo { awsSqsActivityClient.countAllMessagesOnQueue(activityQueueUrl).get() } matches { it == 0 }
-    await untilCallTo { activitiesApi.getCountFor("/activities/$activityId") } matches { it == 1 }
-    await untilCallTo { nomisApi.postCountFor("/activities") } matches { it == 1 }
-    nomisApi.verify(
-      WireMock.postRequestedFor(WireMock.urlEqualTo("/activities"))
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisTest.kt:49]: Qualifiers are now redundant

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisTest.kt [hunk 4]
```diff
@@ -15,161 +25,216 @@ import uk.gov.justice.digital.hmpps.prisonertonomisupdate.wiremock.MappingExtens
...
-      """{
-          "nomisCourseActivityId": $courseActivityId,
-          "activityScheduleId": $activityScheduleId,
-          "mappingType": "TYPE"
-        }
-      """.trimIndent(),
-    )
-    nomisApi.stubAllocationCreate(courseActivityId)
+  @Nested
+  inner class UpdateActivitySchedule {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivityToNomisTest.kt:121]: New tests are in this class

**Review by lbennett-moj — APPROVED** [review]
LGTM

**Pull request merged**